### DISCUSSION TITLE
Symfony 5.4.9 relying on PHP 8 function str_starts_with() ?

### DISCUSSION THREAD

**Discussion by schultz-it-solutions** [symfony/symfony#46740] [Q&A] [answered]
Please bear with me, as I just start getting my head around Symfony...

I installed Symfony 5.4.9 on my server (for a specific project - configured with PHP 7.4.x). It seems to be accessible by that project, but I get a confusing error right away:
```
Fatal error: 
Uncaught Error: Call to undefined function Symfony\Component\HttpFoundation\str_starts_with() in ...\Symfony\Component\HttpFoundation\ServerBag.php:32 
Stack trace: 
#0 ...\Symfony\Component\HttpFoundation\Request.php(284): Symfony\Component\HttpFoundation\ServerBag->getHeaders() 
#1 ...\Symfony\Component\HttpFoundation\Request.php(260): Symfony\Component\HttpFoundation\Request->initialize() 
#2 ...\Symfony\Component\HttpFoundation\Request.php(2039): Symfony\Component\HttpFoundation\Request->__construct() 
#3 ...\Symfony\Component\HttpFoundation\Request.php(306): Symfony\Component\HttpFoundation\Request::createRequestFromFactory() 
#4 ...\welcome.php(15): Symfony\Component\HttpFoundation\Request::createFromGlobals() 
#5 {main} thrown in ...\Symfony\Component\HttpFoundation\ServerBag.php on line 32
```

As I understand, Symfony 5.4.x has a minimum requirement for PHP 7.2, and the "str_starts_with()" function in PHP is a new feature for PHP 8.0
So how comes Symfony 5.4.9 is relying on this PHP8.0 function?
Or am I missing something here?

Kind regards
Ruediger Schultz

**Comment by lyrixx** [comment, marked as answer]
Hello,

Indeed [Symfony 5.4.9](https://github.com/symfony/symfony/blob/v5.4.9/src/Symfony/Component/HttpFoundation/ServerBag.php#L32) use `str_starts_with()` 

But the component also [requires](https://github.com/symfony/symfony/blob/v5.4.9/src/Symfony/Component/HttpFoundation/composer.json#L22) the PHP 8.0 Polyfill

So It should not be an issue.

You composer may have 
```json
{"replace": {"symfony/polyfill-php80: "*"}} 
```
in int, (like [here](https://github.com/symfony/skeleton/blob/6.1/composer.json#L51)).

It it's the case, remove this line, and run `composer update`

**Reply by schultz-it-solutions** [reply]
Yes indeed you are right... 
Once I provided polyfill-php80, this is no longer an issue.
Thank you for your help!
